### DISCUSSION TITLE
Possible Performance Issues?

### DISCUSSION THREAD

**Discussion by orawalters** [microfeed/microfeed#77] [Q&A]
I'm dangerously over the alotted CPU for requests. Anyone know why? I originally brought this up [on the Cloudflare forums](https://community.cloudflare.com/t/far-over-cpu-time-limit-for-free-plan-should-i-be-worried/512633)

**Comment by wenbinf** [comment]
Have you seen any errors yet? If your site has very low traffic (a hobby project), then it should be fine. Otherwise, you may need to upgrade to the [$5/month Workers paid plan](https://developers.cloudflare.com/workers/platform/pricing/) which allows longer duration.

A real-world CRUD web app typically involves multiple db operations to handle a web request (plus some other overheads, e.g., dynamic template rendering...). Therefore, it's not easy to fit in the tight duration budget of Cloudflare Workers free plan. 

In addition, microfeed uses Cloudflare's D1, which is still in alpha and may not be fully optimized.

**Comment by jhonsmithy564-design** [comment]
Sounds like hitting the free plan limits. Upgrading to the paid plan usually solves these CPU and duration issues.
